fdbcli/ThrottleCommand.actor.cpp

```diff
@@ -0,0 +1,645 @@
+/*
+ * ThrottleCommand.actor.cpp
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2013-2021 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "fdbcli/fdbcli.actor.h"
+
+#include "fdbclient/IClientApi.h"
+#include "fdbclient/TagThrottle.h"
+#include "fdbclient/Knobs.h"
+#include "fdbclient/SystemData.h"
+#include "fdbclient/CommitTransaction.h"
+
+#include "flow/Arena.h"
+#include "flow/FastRef.h"
+#include "flow/ThreadHelper.actor.h"
+#include "flow/genericactors.actor.h"
+#include "flow/actorcompiler.h" // This must be the last #include.
+
+namespace {
+
+// Helper functions copied from TagThrottle.actor.cpp
+// The only difference is transactions are changed to go through MultiversionTransaction,
+// instead of the native Transaction(i.e., RYWTransaction)
```

Comment:
This makes me a little uneasy given that it's a pretty big surface that doesn't seem like it's going to be under test to the same degree as the analogous code in TagThrottle.actor.cpp. I also have a hunch that the tag throttle functions are going to change a non-trivial amount in the future, so having it duplicated is definitely a cost.

Reply:
Yeah, I totally agree.
Any idea about how to workaround?
One option is to have TagThrottle.actor.cpp to have interface using `IClientApi`, but looks like it's also used in fdbserver side some places, I am not sure whether this is feasible.

Reply:
I added an issue for this #5292 
Will revisit this soon once have cycles.

Reply:
Would it be possible to make transaction as an argument?

Reply:
yeah, #5292 gives an option to have an upper-level interface where we can pass generic database and transaction pointers as arguments for these functions.
For now, it's a bit hard as ThreadSafeApi and NativeAPI/RYW do not fall into the same category and needs to be dealt with differently.